Intermittent failures in io.vertx.ext.dropwizard.MetricsTest.testHttpMetricsResponseCode

```
Starting test: MetricsTest#testHttpMetricsResponseCode 
java.lang.AssertionError: Was expecting responses-3xx to be not null
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertNotNull(Assert.java:712)
	at io.vertx.test.core.AsyncTestBase.assertNotNull(AsyncTestBase.java:390)
	at io.vertx.ext.dropwizard.MetricsTest.lambda$test$30(MetricsTest.java:310)
	at io.vertx.ext.dropwizard.MetricsTest$$Lambda$254/1687627235.handle(Unknown Source)
	at io.vertx.ext.dropwizard.MetricsTest.lambda$test$32(MetricsTest.java:327)
	at io.vertx.ext.dropwizard.MetricsTest$$Lambda$256/1276544608.handle(Unknown Source)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$14(HttpServerImpl.java:296)
	at io.vertx.core.http.impl.HttpServerImpl$$Lambda$79/220813571.handle(Unknown Source)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$3(ContextImpl.java:359)
	at io.vertx.core.impl.ContextImpl$$Lambda$34/1422238463.run(Unknown Source)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:339)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:374)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:742)
	at java.lang.Thread.run(Thread.java:745)
Unhandled exception 
java.lang.AssertionError: Was expecting responses-3xx to be not null
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertNotNull(Assert.java:712)
	at io.vertx.test.core.AsyncTestBase.assertNotNull(AsyncTestBase.java:390)
	at io.vertx.ext.dropwizard.MetricsTest.lambda$test$30(MetricsTest.java:310)
	at io.vertx.ext.dropwizard.MetricsTest$$Lambda$254/1687627235.handle(Unknown Source)
	at io.vertx.ext.dropwizard.MetricsTest.lambda$test$32(MetricsTest.java:327)
	at io.vertx.ext.dropwizard.MetricsTest$$Lambda$256/1276544608.handle(Unknown Source)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$14(HttpServerImpl.java:296)
	at io.vertx.core.http.impl.HttpServerImpl$$Lambda$79/220813571.handle(Unknown Source)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$3(ContextImpl.java:359)
	at io.vertx.core.impl.ContextImpl$$Lambda$34/1422238463.run(Unknown Source)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:339)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:374)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:742)
	at java.lang.Thread.run(Thread.java:745)

java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at io.vertx.test.core.AsyncTestBase.assertTrue(AsyncTestBase.java:372)
	at io.vertx.test.core.AsyncTestBase.awaitLatch(AsyncTestBase.java:592)
	at io.vertx.ext.dropwizard.MetricsTest.test(MetricsTest.java:329)
	at io.vertx.ext.dropwizard.MetricsTest.testHttpMetricsResponseCode(MetricsTest.java:297)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:283)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:173)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:128)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
```

--- a/src/main/asciidoc/groovy/index.adoc
+++ b/src/main/asciidoc/groovy/index.adoc
@@ -22,7 +22,7 @@ To enable metrics, add the following dependency to the _dependencies_ section of
 <dependency>
   <groupId>io.vertx</groupId>
   <artifactId>vertx-dropwizard-metrics</artifactId>
-  <version>3.3.3</version>
+  <version>3.4.0-SNAPSHOT</version>
 </dependency>
 ----
 
@@ -30,7 +30,7 @@ To enable metrics, add the following dependency to the _dependencies_ section of
 
 [source,groovy,subs="+attributes"]
 ----
-compile 'io.vertx:vertx-dropwizard-metrics:3.3.3'
+compile 'io.vertx:vertx-dropwizard-metrics:3.4.0-SNAPSHOT'
 ----
 
 Then when you create vertx enable metrics using the `link:../dataobjects.html#DropwizardMetricsOptions[DropwizardMetricsOptions]`:

--- a/src/main/asciidoc/java/index.adoc
+++ b/src/main/asciidoc/java/index.adoc
@@ -22,7 +22,7 @@ To enable metrics, add the following dependency to the _dependencies_ section of
 <dependency>
   <groupId>io.vertx</groupId>
   <artifactId>vertx-dropwizard-metrics</artifactId>
-  <version>3.3.3</version>
+  <version>3.4.0-SNAPSHOT</version>
 </dependency>
 ----
 
@@ -30,7 +30,7 @@ To enable metrics, add the following dependency to the _dependencies_ section of
 
 [source,groovy,subs="+attributes"]
 ----
-compile 'io.vertx:vertx-dropwizard-metrics:3.3.3'
+compile 'io.vertx:vertx-dropwizard-metrics:3.4.0-SNAPSHOT'
 ----
 
 Then when you create vertx enable metrics using the `link:../../apidocs/io/vertx/ext/dropwizard/DropwizardMetricsOptions.html[DropwizardMetricsOptions]`:

--- a/src/main/asciidoc/js/index.adoc
+++ b/src/main/asciidoc/js/index.adoc
@@ -22,7 +22,7 @@ To enable metrics, add the following dependency to the _dependencies_ section of
 <dependency>
   <groupId>io.vertx</groupId>
   <artifactId>vertx-dropwizard-metrics</artifactId>
-  <version>3.3.3</version>
+  <version>3.4.0-SNAPSHOT</version>
 </dependency>
 ----
 
@@ -30,7 +30,7 @@ To enable metrics, add the following dependency to the _dependencies_ section of
 
 [source,groovy,subs="+attributes"]
 ----
-compile 'io.vertx:vertx-dropwizard-metrics:3.3.3'
+compile 'io.vertx:vertx-dropwizard-metrics:3.4.0-SNAPSHOT'
 ----
 
 Then when you create vertx enable metrics using the `link:../dataobjects.html#DropwizardMetricsOptions[DropwizardMetricsOptions]`:

--- a/src/main/asciidoc/ruby/index.adoc
+++ b/src/main/asciidoc/ruby/index.adoc
@@ -22,7 +22,7 @@ To enable metrics, add the following dependency to the _dependencies_ section of
 <dependency>
   <groupId>io.vertx</groupId>
   <artifactId>vertx-dropwizard-metrics</artifactId>
-  <version>3.3.3</version>
+  <version>3.4.0-SNAPSHOT</version>
 </dependency>
 ----
 
@@ -30,7 +30,7 @@ To enable metrics, add the following dependency to the _dependencies_ section of
 
 [source,groovy,subs="+attributes"]
 ----
-compile 'io.vertx:vertx-dropwizard-metrics:3.3.3'
+compile 'io.vertx:vertx-dropwizard-metrics:3.4.0-SNAPSHOT'
 ----
 
 Then when you create vertx enable metrics using the `link:../dataobjects.html#DropwizardMetricsOptions[DropwizardMetricsOptions]`:

--- a/src/test/java/io/vertx/ext/dropwizard/MetricsTest.java
+++ b/src/test/java/io/vertx/ext/dropwizard/MetricsTest.java
@@ -293,10 +293,10 @@ public class MetricsTest extends MetricsTestBase {
     test(300, "responses-3xx");
     test(404, "responses-4xx");
     test(500, "responses-5xx");
-
   }
 
   private void test(int code, String metricName) throws Exception {
+    CountDownLatch closeLatch = new CountDownLatch(2);
     CountDownLatch latch = new CountDownLatch(1);
     HttpClient client = vertx.createHttpClient(new HttpClientOptions());
     Handler<HttpServer> doTest = (server) -> {
@@ -314,10 +314,14 @@ public class MetricsTest extends MetricsTestBase {
           }
           latch.countDown();
         });
+      }).connectionHandler(connection -> {
+        connection.closeHandler(v -> closeLatch.countDown());
       }).end();
     };
     HttpServer server = vertx.createHttpServer(new HttpServerOptions().setHost("localhost").setPort(8080)).requestHandler(req -> {
       req.response().setStatusCode(code).end();
+    }).connectionHandler(connection -> {
+      connection.closeHandler(v -> closeLatch.countDown());
     }).listen(ar -> {
       assertTrue(ar.succeeded());
       doTest.handle(ar.result());
@@ -325,6 +329,7 @@ public class MetricsTest extends MetricsTestBase {
     awaitLatch(latch);
     cleanup(client);
     cleanup(server);
+    awaitLatch(closeLatch);
   }
 
   @Test

